commitizen/version_providers.py

```diff
@@ -0,0 +1,38 @@
+from packaging.version import Version
+
+
+class SemVerVersion(Version):
```

Comment:
This is very clever. In my initial work I was thinking of creating a protocol for versions, like `VersionProtocol` and using [semver](https://pypi.org/project/semver/). 
1. Do you think re-implementing `__str__` could create any issues in the long term?
2. If we had to implement another version_type, let's say [calver](https://calver.org/) what should users implement?

Reply:
Look here please.
Only __str__ did not work out, I added to the protocol all the necessary methods that are used.

https://github.com/commitizen-tools/commitizen/pull/686/commits/f057e3f20cc026887a0d436922385a09f1af40ac#diff-ebe7b665130281170ae6aa39905fc0b3ba04a59b0d98787509c5aa3fce1b1b28R12